cumulusci/tasks/release_notes/task.py

```diff
@@ -55,3 +73,127 @@ def _run_task(self):
 
         release_notes = generator()
         self.logger.info("\n" + release_notes)
+
+
+class ParentPullRequestNotes(BaseGithubTask):
+    task_docs = """
+    Aggregate change notes from child pull request(s) to its corresponding
+    parent's pull request.
+    
+    When given the branch_name option, this task will: (1) check if the base branch
+    of the corresponding pull request starts with the feature branch prefix and if so (2) attempt
+    to query for a pull request corresponding to this parent feature branch. (3) if a pull request
+    isn't found, one is created and the BUILD_NOTES_LABEL is applied to it.
+
+    If the BUILD_NOTES_LABEL is present on the pull request, then all notes from the 
+    child pull request are aggregated into the parent pull request. If the BUILD_NOTES_LABEL
+    is not detected on the parent pull request then a link to the child pull request
+    is placed under the "Unaggregated Pull Reqeusts" header.
+
+    When given the parent_branch_name option, this task will query for a corresponding pull request.
+    If a pull request is not found, the task exits. If a pull request is found, then all notes
+    from child pull requests are re-aggregated and the body of the parent is replace entirely.
+    """
+
+    BUILD_NOTES_LABEL = "Build Change Notes"
+
+    task_options = {
+        "branch_name": {"description": "Name of branch with a pull request"},
+        "parent_branch_name": {
+            "description": "name of the parent branch to rebuild change notes for"
+        },
+    }
+
+    def _run_task(self):
+        branch_name = self.options.get("branch_name")
+        parent_branch_name = self.options.get("parent_branch_name")
+
+        if (not branch_name and not parent_branch_name) or (
+            branch_name and parent_branch_name
+        ):
+            raise TaskOptionsError(
+                "You must specify either branch_name or (exclusive) parent_branch_name."
+            )
```

Comment:
Option validation should go in the task's _init_options method. (That way it's raised when the task is constructed instead of when it's running.)